### PR TITLE
Add new function arrayZipUnaligned

### PR DESCRIPTION
**Pull request by taiyang-li**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add new function `arrayZipUnaligned` for spark compatiablity(arrays_zip), which allowed unaligned arrays based on original `arrayZip`. 
``` sql
SELECT arrayZipUnaligned([1], [1, 2, 3])

Query id: 3ecfa2db-9608-4b39-9953-bb794fe8eae0

   ┌─arrayZipUnaligned([1], [1, 2, 3])─┐
1. │ [(1,1),(NULL,2),(NULL,3)]         │
   └───────────────────────────────────┘

1 row in set. Elapsed: 0.002 sec. 
```

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 5b3ca6b2b916b63d12ddb402c0726d0c2e29c1b1 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69030/5b3ca6b2b916b63d12ddb402c0726d0c2e29c1b1/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by fm4v** [comment]
Hi! Tests missing

**Comment by taiyang-li** [comment]
> Hi! Tests missing

Sorry, I'll turn it into draft status before I'm ready.

**Comment by taiyang-li** [comment]
@fm4v this pr is ready for review now

**Comment by fm4v** [comment]
Fuzzer error https://s3.amazonaws.com/clickhouse-test-reports/69030/bd0ddf85eb987517e99edc467a2206ad4033f64f/ast_fuzzer__asan_.html

```
SELECT DISTINCT concat(concat(concat(arrayZipUnaligned(materialize(['g', 'h', 'i'])), (1, 2, 3), 1, ('a', 'b', 'c')), arrayZipUnaligned(['a']), 2), (1, 'y')) FROM system.parts
```

**Comment by taiyang-li** [comment]
failed ut not related to this pr : 
image

File: src/Functions/array/arrayZip.cpp [hunk 1]
```diff
@@ -39,65 +39,149 @@ class FunctionArrayZip : public IFunction
...
+                    ErrorCodes::SIZES_OF_ARRAYS_DONT_MATCH,
+                    "The argument 1 and argument {} of function {} have different array sizes",
+                    unaligned_index + 1,
+                    getName());
+            else
+                return ColumnArray::create(
+                    ColumnTuple::create(std::move(tuple_columns)), static_cast<const ColumnArray &>(*holders[0]).getOffsetsPtr());
+        }
+        else
+            return executeUnaligned(holders, tuple_columns, input_rows_count);
```
Comment (CurtizJ) [inline src/Functions/array/arrayZip.cpp:115]: It's possible to execute lightweight version of this function if there are no unaligned arrays: just wrap tuple columns into nullable.

**Review by CurtizJ — COMMENTED** [review]
In general ok.

File: src/Functions/array/arrayZip.cpp (hunk 1 above)
Comment (taiyang-li) [reply src/Functions/array/arrayZip.cpp]: fixed.

**Comment by taiyang-li** [comment]
Failed performance test that is identical with last failure is not related to this pr

**Review by CurtizJ — APPROVED** [review]

**Pull request merged**